src/app/users-page/author/add-authors/add-authors.component.spec.ts

```diff
@@ -32,21 +47,36 @@ describe('AddAuthorsComponent', () => {
     fixture.detectChanges();
   });
 
-  // it('form valid when input is entered', () => {
-  //   const firstNameInput = authorComponent.registerFormControl.first_name;
-  //   firstNameInput.setValue('Delyan');
+  it('should make field required', () => {
+    const firstNameInput =
+      authorComponent.registerAuthorForm.controls.first_name;
+    expect(firstNameInput.errors.required).toBeTruthy();
 
-  //   const lastNameInput = authorComponent.registerFormControl.last_name;
-  //   lastNameInput.setValue('Georgiev');
+    firstNameInput.setValue('John');
+    expect(firstNameInput.errors).toBeNull();
+  });
+
+  it('form should be invalid when empty', () => {
+    authorComponent.registerAuthorForm.controls.first_name.setValue('');
+    authorComponent.registerAuthorForm.controls.last_name.setValue('');
+    expect(authorComponent.registerAuthorForm.valid).toBeFalsy();
+  });
 
-  //   expect(authorComponent.registerAuthorForm.valid).toBeTruthy();
-  // });
+  it('form valid when input is entered', () => {
+    const firstNameInput = authorComponent.registerFormControl.first_name;
+    firstNameInput.setValue('Delyan');
 
-  // it('should test input errors', () => {
-  //   const firstNameInput = authorComponent.registerFormControl.first_name;
-  //   expect(firstNameInput.errors.required).toBeTruthy();
+    const lastNameInput = authorComponent.registerFormControl.last_name;
+    lastNameInput.setValue('Georgiev');
 
-  //   firstNameInput.setValue('Delyan');
-  //   expect(firstNameInput.errors).toBeNull();
-  // });
+    expect(authorComponent.registerAuthorForm.valid).toBeTruthy();
+  });
+
+  it('should test input errors', () => {
```

Comment:
I don't think this test is very different from `it('form valid when input is entered', ())`

Reply:
Consider also testing the **addAuthor** and **onSubmit** methods.